recipes/couchdb/build.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+set -eux
+
+if [[ $(uname) == Darwin ]]; then
+    sed -i '' "s|-I/usr/local/opt/icu4c/include -I/opt/homebrew/opt/icu4c/include|-I$PREFIX/include|g" src/couch/rebar.config.script
+    sed -i '' "s|-L/usr/local/opt/icu4c/lib -L/opt/homebrew/opt/icu4c/lib|-L$PREFIX/lib|g" src/couch/rebar.config.script
+fi
+export CFLAGS="-I$PREFIX/include -I$PREFIX/include/mozjs-91 -I$PREFIX/lib/erlang/usr/include"
+export LDFLAGS="-L$PREFIX/lib"
+export ERL_CFLAGS="$CFLAGS"
+export ERL_LDFLAGS="$LDFLAGS"
+./configure --erlang-md5 --spidermonkey-version 91
+make release
+
+install -dm755 $PREFIX/lib
+cp -r rel/couchdb $PREFIX/lib/couchdb
```

Comment:
Why are these not installed directly to `$PREFIX/bin/`?

Reply:
That's the way it's done in pretty much every distro package of couchdb out there, and to my understanding just how Erlang applications are distributed in general.

As noted above, when installing _everything_ directly to the prefix, it results in conflicts between bundled files that are also present in other packages. The main couchdb binary is actually another script that uses relative paths to launch the bundled erlang runtime with all the correct options, so moving the binary outside the couchdb bundle will not work: https://github.com/apache/couchdb/blob/main/rel/files/couchdb.in

In fact in most distro packages couchdb isn't even on the path, and is directly launched inside its bundle by whatever init system the distro uses: https://github.com/archlinux/svntogit-community/blob/421c3a4bb3d42c33335048635bebcd191f29f62a/trunk/couchdb.service#L11 

Since conda doesn't have an init system couchdb has to be started in another way, which is why I've added a wrapper script.

Reply:
@isuruf does this make sense? I feel like we're going in circles since this is pretty much the issue @carterbox [raised above](https://github.com/conda-forge/staged-recipes/pull/18302#pullrequestreview-975746465).